elisp/mew-auth.el

```diff
@@ -67,6 +67,80 @@
 (defun mew-keyed-md5 (key passwd)
   (mew-md5 (concat key passwd)))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; XOAUTH2
+
+(defun mew-auth-xoauth2-auth-string (user token)
+  ;; base64(user=user@example.com^Aauth=Bearer ya29vF9dft4...^A^A)
+  (base64-encode-string (format "user=%s\1auth=Bearer %s\1\1" user token) t))
+
+(defun mew-auth-xoauth2-json-status (status-string)
+  ;; https://developers.google.com/gmail/imap/xoauth2-protocol#error_response_2
+  (require 'json)
+  (let ((json-status
+         (ignore-errors
+           (json-read-from-string
+            (base64-decode-string status-string)))))
+    (if json-status
+        (if (string-match "^2" (cdr (assoc 'status json-status)))
+            "OK" ;; 2XX
+          "NO" ;; XXX: Anyway NO?
```

Comment:
This line shoud be `"OK") ;; 2XX` and the next line should be removed.

Reply:
Let me clarify my understanding.
"This line" means line 86 or 87?
If 86, we can not have a chance to return "NO" on failure.

Gmail seems not to return JSON string on success.
So, I want to return *"OK"* if `json-string` is nil.
If `json-string` is valid, I want to respect the `status` value in it.

Reply:
I don't talk about logic, but talk about coding-style.
Closing brackets must not consume a line by themselves.
This is *the* custom shared by many Lispers.

Reply:
Ah..., thanks. I'll remove some newlines.

To tell the truth, Line 88  and 89 should have had some code to take care other OAuth providers such as Office 365.
It was the lingering newline of my hesitation :sweat_smile: 

By the way, is there any handy way for me to repair this PR authored by @tats san?
If not, do you mind merging this PR with the unwilling nelines?
Following this PR, I'll make a new PR that removes newlines.